src/com/owncloud/android/services/observer/InstantUploadsObserverBasedOnCommonsIOFileMonitor.java

```diff
@@ -0,0 +1,203 @@
+     * @param configuration     New configuration for instant uploads to replace the current one.
+     * @return                  'True' if the new configuration could be handled by the observer,
+     *                          'false' otherwise.
+     */
+    @Override
+    public boolean updateConfiguration(InstantUploadsConfiguration configuration) {
+        if (configuration == null) {
+            return false;
+        }
+        synchronized (mLock) {
+            if (mConfiguration.getSourcePath().equals(configuration.getSourcePath())) {
+                mConfiguration = configuration;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void startObserving() {
+        try {
+            initialize();
+            addListener(new CameraFolderAlterationListener());
+            mMonitor.addObserver(this);
+            mMonitor.start();
+
+        } catch (Exception e) {
+            Log_OC.e(
+                TAG,
+                "Exception starting to watch camera folder, instant uploads will not work",
+                e
+            );
+        }
+    }
+
+    @Override
+    public void stopObserving() {
+        try {
+            mMonitor.stop();
+            mMonitor.removeObserver(this);
+            checkAndNotify();
+            destroy();
+
+        } catch (Exception e) {
+            Log_OC.w(
+                TAG,
+                "Exception stopping to watch camera folder: " + e.getMessage()
+            );
+        }
+
+    }
+
+    private class CameraFolderAlterationListener implements FileAlterationListener {
+
+        private static final int HANDLE_DELAY_IN_MS = 200;
+
+        private Handler mHandler = new Handler();
+
+        @Override
+        public void onStart(FileAlterationObserver observer) {
+            Log_OC.v(TAG, "onStart called");
+        }
+
+        @Override
+        public void onDirectoryCreate(File directory) {
+            Log_OC.i(TAG, "onDirectoryCreate called for " + directory.getAbsolutePath());
+        }
+
+        @Override
+        public void onDirectoryChange(File directory) {
+            Log_OC.i(TAG, "onDirectoryChange called for " + directory.getAbsolutePath());
+        }
+
+        @Override
+        public void onDirectoryDelete(File directory) {
+            Log_OC.i(TAG, "onDirectoryDelete called for " + directory.getAbsolutePath());
+        }
+
+        @Override
+        public void onFileCreate(final File file) {
+            if (file != null) {
+                Log_OC.i(TAG, "onFileCreate called for " + file.getAbsolutePath());
+                synchronized (mLock) {
+                    final String fileName = file.getAbsolutePath().substring(
+                        mConfiguration.getSourcePath().length() + 1
+                    );
+
+                    /// delay a bit the execution to deal with possible renames of files (for instance: Google Camera)
+                    mHandler.postDelayed(
```

Comment:
I don't fully understand the delay that is being used here just to handle file renaming. Does the Google Camera change the file name after creating it? If so, shouldn't this case be managed in onFileChange method?